clang/include/clang/Basic/AttrDocs.td

```diff
@@ -3967,6 +3967,69 @@ Attribute ``trivial_abi`` has no effect in the following cases:
   }];
 }
 
+
+def LifetimeCaptureByDocs : Documentation {
+  let Category = DocCatFunction;
+  let Content = [{
+    The ``lifetime_capture_by(X)`` attribute on a function parameter or implicit object
+parameter indicates that references to arguments passed to such parameters may be
+captured by the capturing entity ``X``.
+
+The capturing entity ``X`` can be one of the following:
+- Another (named) function parameter. 
+  
+  .. code-block:: c++
+
+    void addToSet(std::string_view a [[clang::lifetime_capture_by(s)]], std::set<std::string_view>& s) {
+      s.insert(a);
+    }
+
+- ``this`` (in case of member functions).
+  
+  .. code-block:: c++
+
+    class S {
+      void addToSet(std::string_view a [[clang::lifetime_capture_by(this)]]) {
+        s.insert(a);
+      }
+      std::set<std::string_view> s;
+    };
+
+- 'global', 'unknown' (without quotes).
+  
+  .. code-block:: c++
+
+    std::set<std::string_view> s;
+    void addToSet(std::string_view a [[clang::lifetime_capture_by(global)]]) {
```

Comment:
One question is, how would this behave when there is an identifies collision. E.g.:
```
void addToSet(std::string_view a [[clang::lifetime_capture_by(global)]], bool global)
```
?

Reply:
Hmm. I didn't think about this before. Would switching to something like `__global`, `__unknown` make sense ?

Reply:
I am ok with either using the `__` names or just documenting the behavior and a workaround (user renaming their arguments). Alternatively, we could also permit users to write indices instead of names. I don't have a really strong preference.

Reply:
I wonder if @erichkeane has an opinion on this.

Reply:
I think it is an ambiguous case, we could emit a diagnostic.